kotest-assertions/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/iterator/IteratorMatchersTest.kt

```diff
@@ -0,0 +1,45 @@
+package com.sksamuel.kotest.matchers.iterator
+
+import io.kotest.matchers.iterator.shouldBeEmpty
+import io.kotest.matchers.iterator.shouldNotBeEmpty
+import io.kotest.specs.WordSpec
+
+class IteratorMatchersTest: WordSpec() {
+
+   init {
+
+      "shouldBeEmpty" should {
```

Comment:
I would suggest that instead of testing with your own implementation of `Iterator`, use ones available.

Something like `listOf<Int>().iterator().shouldBeEmpty()`. Your implementation of iterator, although tests what we want, is incorrect.

If we change the implementation in the future, this test will crash, and I don't think we want that.

What do you think?

Reply:
Sounds good. I keep forgetting some of these are already defined :-/